tests/unittests/tests-core/tests-core-clist.c

```diff
@@ -22,46 +22,149 @@ static list_node_t test_clist;
 static void set_up(void)
 {
     memset(tests_clist_buf, 0, sizeof(tests_clist_buf));
+    test_clist.next = NULL;
 }
 
-static void test_clist_add_one(void)
+static void test_clist_rpush(void)
 {
+    set_up();
     list_node_t *elem = &(tests_clist_buf[0]);
     list_node_t *list = &test_clist;
-    list->next = NULL;
 
-    clist_insert(list, elem);
+    clist_rpush(list, elem);
 
     TEST_ASSERT_NOT_NULL(list->next);
     TEST_ASSERT(list->next->next == list->next);
 }
 
 static void test_clist_add_two(void)
 {
+    set_up();
```

Comment:
nope, this function is called from the other tests, and they expect a defined state.